File: Code-spark/services/common-node/src/dto/ApiResponse.js
class ApiResponse {
  constructor({ success, message, data, error }) {
    this.success = success;
    if (message !== undefined) this.message = message;
    if (data !== undefined) this.data = data;
    if (error !== undefined) this.error = error;
  }

  static success(dataOrMessage, data = null) {
    if (typeof dataOrMessage === 'string') {
      return new ApiResponse({ success: true, message: dataOrMessage, data });
    }
    return new ApiResponse({ success: true, data: dataOrMessage });
  }

  static error(message, error = null) {
    return new ApiResponse({ success: false, message, error });
  }
}

module.exports = ApiResponse;

File: Code-spark/services/notification-service/src/main/java/com/codespark/notificationservice/dto/ApiResponse.java
package com.codespark.notificationservice.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Builder;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
@Builder
public class ApiResponse<T> {
    private boolean success;
    private String message;
    private T data;

    public boolean isSuccess() {
        return success;
    }

    public void setSuccess(boolean success) {
        this.success = success;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }
}
